### PR TITLE
Allow column_index_size_in_kb to be configurable through nodetool

### PR DESCRIPTION
**Pull request by frankgh**

### CODE REVIEW

File: src/java/org/apache/cassandra/service/StorageService.java [hunk 1]
```diff
@@ -5833,6 +5830,18 @@ public void setCachedReplicaRowsFailThreshold(int threshold)
...
+    public int getColumnIndexSizeInKB()
+    {
+        return DatabaseDescriptor.getColumnIndexSizeInKB();
+    }
+
+    public void setColumnIndexSize(int columnIndexSizeInKB)
+    {
+        int oldValueInKB = DatabaseDescriptor.getColumnIndexSizeInKB();
+        DatabaseDescriptor.setColumnIndexSize(columnIndexSizeInKB);
+        logger.info("setColumnIndexSize: column index size set to {} KiB (was {} KiB)", columnIndexSizeInKB, oldValueInKB);
```
Comment (yifan-c) [inline src/java/org/apache/cassandra/service/StorageService.java]: How about "Updated column_index_size_in_kb to ..."? The other methods log the message in this format.

File: src/java/org/apache/cassandra/tools/nodetool/GetColumnIndexSize.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package org.apache.cassandra.tools.nodetool;
+
+import io.airlift.airline.Command;
+import org.apache.cassandra.tools.NodeProbe;
+import org.apache.cassandra.tools.NodeTool.NodeToolCmd;
+
+@Command(name = "getcolumnindexsize", description = "Print the granularity of the collation index of rows within a partition in KiB")
+public class GetColumnIndexSize extends NodeToolCmd
```
Comment (yifan-c) [inline src/java/org/apache/cassandra/tools/nodetool/GetColumnIndexSize.java:26]: Personally, adding 2 nodetool commands just for getter and setter is an overkill. But there are existing nodetools already doing so. 🤷

**Review by yifan-c — COMMENTED** [review]
Please remove the annotation `@VisibleForTesting` for `org.apache.cassandra.config.DatabaseDescriptor#setColumnIndexSize`. The annotation is used as "only for testing".

Making the configuration a dynamic one means the width of `IndexInfo` can be variable for a single SSTable. It should be fine as the width is encoded for each index info, so no issue with reading.

**Comment by frankgh** [comment]
@yifan-c thanks for the review. I have addressed your comments.

**Pull request closed without merging**